### ISSUE TITLE
tf2.1.0 training efficiency problem 

### ISSUE THREAD

**Issue by chenpengf0223** [tensorflow/tensorflow#40980]
**System information**
- Have I written custom code (as opposed to using a stock example script provided in TensorFlow):
- OS Platform and Distribution: Linux Ubuntu 16.04
- TensorFlow installed from (source or binary): binary
- TensorFlow version (use command below): 2.1.0
- Python version: 3.7.7 
- CUDA/cuDNN version: 10.1.0/7.6.4
- GPU model and memory: Tesla V100*4  32478MiB

**Describe the current behavior**
I use tf2.1.0 keras API to build network and use tf API to build a semantic segmentation distillation loss. And I find that the training speed is about 42.1 epochs/day. I try to adjust the "batch size"，but the result is almost the same.

If I use tf1.10.0 tf API to build network and loss,  the training speed is about 134  epochs/day.

**Describe the expected behavior**
TF2.1.0 Training speed is not lower than tf1.10.0.

**Standalone code to reproduce the issue**
I call function "pairwise_suprvise" to calculate my loss(The student_feature/teacher_feature shape is [N, 512, 512, 3]):

My code is linked:  https://github.com/chenpengf0223/semantic_segmentation_distillation/blob/master/distillation_loss.py

**Comment by amahendrakar** [comment]
@chenpengf0223,
In the code sample, you have provided the function definitions but they are not being called anywhere. Could you please provide the complete code to reproduce the issue reported here.

Also, please upgrade to the latest version of TensorFlow 2.2 and check if you are facing the same issue. Thanks!

**Comment by chenpengf0223** [comment]
@amahendrakar ,thanks for your apply, I call the function "pairwise_suprvise" like this:
 RKD_dist_loss, RKD_angle_loss = pairwise_suprvise(feature_map_student, feature_map_teacher, accessible_area_distillation_mask)

The shape of "feature_map_student" and "feature_map_teacher" is [N, 512, 512, 3].

I find that the training will be 3x times faster if do not call the "pairwise_suprvise" function.

**Comment by chenpengf0223** [comment]
@amahendrakar , I try tf2.2.0，the performance is almost the same with tf1. So maybe there is little problem in tf2.1.0. Thanks for your guidance.

**Comment by amahendrakar** [comment]
> When I set batch_size = 28, every gpu memory occupation was about 9G, and when I set batch_size = 32, the training crashed. It seemed that gpu memory was not enough.

@chenpengf0223,
Please take a look at [this guide](https://www.tensorflow.org/guide/gpu#limiting_gpu_memory_growth) to limit GPU memory growth and let us know if it helps. Thanks!

**Comment by chenpengf0223** [comment]
@amahendrakar  Thanks for reply.
I have checked my code, and found that I already limited GPU memory growth as following:

for gpu in gpus:
    tf.config.experimental.set_memory_growth(device=gpu, enable=True)

When I set batch_size = 28, every gpu memory occupation was about 9G, If I do not limit GPU memory growth, the occupation would be 32G, right?

**Comment by amahendrakar** [comment]
> When I set batch_size = 28, every gpu memory occupation was about 9G, If I do not limit GPU memory growth, the occupation would be 32G, right?

@chenpengf0223,
By default, TensorFlow maps nearly all of the GPU memory of all GPUs visible to the process.

Try setting a hard limit on the total memory to allocate on the GPU using `tf.config.experimental.set_virtual_device_configuration` as shown in [this guide](https://www.tensorflow.org/guide/gpu#limiting_gpu_memory_growth) and let us know if it helps. Thanks!
